### PR TITLE
chore(external docs): Add `rename_fields` transform metadata

### PR DESCRIPTION
**Pull request by binarylogic**
Completes https://github.com/timberio/vector/issues/4303 for `rename_fields`

### CODE REVIEW

File: docs/reference/components/transforms/rename_fields.cue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  support: {
+      input_types: ["log"]
+
+    platforms: {
+      "aarch64-unknown-linux-gnu": true
+      "aarch64-unknown-linux-musl": true
+      "x86_64-apple-darwin": true
+      "x86_64-pc-windows-msv": true
```
Comment (Hoverbear) [inline docs/reference/components/transforms/rename_fields.cue:30]:
```suggestion
      "x86_64-pc-windows-msvc": true
```

File: docs/reference/components/transforms/rename_fields.cue [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package metadata
+
+components: transforms: rename_fields: {
+  title: "#{component.title}"
```
Comment (Hoverbear) [inline docs/reference/components/transforms/rename_fields.cue:4]:
```suggestion
  title: "Rename Fields"
```

**Review by Hoverbear — APPROVED** [review]
bors r+

**Comment by binarylogic** [comment]
@Hoverbear same here. Please see #4303 on what needs to be done. Let me know if you have any questions.

File: docs/reference/components/transforms/rename_fields.cue [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      body: #"""
+            Keys specified in this transform will replace existing keys.
+            """#
+      sub_sections: [
+        {
+          title: "Key Conflicts"
+          body: #"""
+                Keys specified in this transform will replace existing keys.
+
+                <Alert type="warning">
```
Comment (Hoverbear) [inline docs/reference/components/transforms/rename_fields.cue:74]: @binarylogic Is this the correct way to embed the alerts?

**Pull request merged**